logtail/filch/openfilesync_darwin.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) 2021 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//+build darwin
+
+package filch
+
+import (
+	"fmt"
+	"os"
+	"syscall"
+
+	"golang.org/x/sys/unix"
+)
+
+func openFileSync(path string, flag int, perm os.FileMode) (*os.File, error) {
+	f, err := os.OpenFile(path, flag, perm)
+	if err != nil {
+		return nil, err
+	}
+	_, err = unix.FcntlInt(uintptr(f.Fd()), syscall.F_NOCACHE, 1)
```

Comment:
using `(*os.File).Fd` has weird side effects. You should use `(*os.File).SyscallConn` these days instead I think.